test/Elastic.Apm.Tests/StaticTests.cs

```diff
@@ -0,0 +1,25 @@
+using FluentAssertions;
+using Xunit;
+
+namespace Elastic.Apm.Tests
+{
+	/// <summary>
+	/// These tests access the static <see cref="Agent"/> instance.
+	/// All other tests should have their own <see cref="ApmAgent"/> instance and not rely on anything static.
+	/// Tests accessing the static <see cref="Agent"/> instance can not run in parallel with tests that also access the static instance.
```

Comment:
BTW the problem is not only running tests using agent singleton in parallel - even when run sequentially the tests can affect each other because the tests (at least the one below) don't reset changes to the global state.

Reply:
Right, the order also matters. I really hope we won't have more such tests. We can control the ordering easily.